drivers/gles2/rasterizer_scene_gles2.cpp

```diff
@@ -1133,8 +1133,8 @@ void RasterizerSceneGLES2::_add_geometry_with_material(RasterizerStorageGLES2::G
 
 				LightInstance *li = light_instance_owner.getornull(e->instance->light_instances[i]);
 
-				if (li->light_index >= render_light_instance_count) {
-					continue; // too many
+				if (render_light_instances[li->light_index] != li || li->light_index >= render_light_instance_count) {
```

Comment:
The order of the bools should be reversed. Otherwise when ``light_index >= render_light_instance_count``, ``render_light_instances[li->light_index]`` will crash because its an out of bounds error  

``if (li->light_index >= render_light_instance_count || render_light_instances[li->light_index] != li)``

Reply:
Will make those changes later today. I was thinking about adding a todo comment regarding what you said in your first comment - should I add it?

Reply:
No, I dont think so. That was more a note to reduz/akien. We may want to just apply this fix and not worry about the underlying problem.as its gonna be re-written for 4.0 anyway.

Reply:
Condition order changed.